### DISCUSSION TITLE
Even if it is opened in a location other than root, I want to display it from root.

### DISCUSSION THREAD

**Discussion by ya2s** [nvim-tree/nvim-tree.lua#2778] [Q&A]
Even if I open it in a location other than root, I want it to be displayed from root.
Now I can achieve this by introducing project.nvim
Is it possible to replace this with vim.fs.root() from 0.10?

**Comment by ya2s** [comment]
```
  vim.api.nvim_create_autocmd("BufEnter", {
    callback = function(ctx)
      local root = vim.fs.root(ctx.buf, { ".git" })
      if root then
        vim.uv.chdir(root)
      end
    end,
  })
```
I was able to achieve this, but if there is another way please let me know

**Reply by alex-courtis** [reply]
That looks pretty good to me; nvim-tree doesn't have a mechanism to do this.
